### ISSUE TITLE
Support for Kafka Message Broker

### ISSUE THREAD

**Issue by vanthome** [tcdl/msb#87]
Hi,
I really like the design of this lib.
Have you though about supporting Apache Kafka as Broker?
BG

**Comment by galkin** [comment]
Hello, vanthome.
My teams don't have need for Apache Kafka as Broker. If you want make pull request in lib/adapters.

**Comment by jupiter** [comment]
Hi @vanthome, we did originally have a Kafka (0.8) broker, but we removed it due to the way Kafka handled topic subscription. If I remember correctly, if you unsubscribed from Kafka, it would take 30 seconds before a new subscriber would start receiving the messages that would have been intended for the previous subscriber. I think Zookeeper was a factor in this problem. It just didn't work well with the way MSB configures subscriptions dynamically. However, this behaviour was supposed to have improved with later versions of Kafka, so it may be worth revisiting.
